src/framework/theme/components/cdk/overlay/overlay-trigger.ts

```diff
@@ -188,12 +188,18 @@ export class NbFocusTriggerStrategy extends NbTriggerStrategy {
     .pipe(takeWhile(() => this.isHostInBody()));
 }
 
-export class NbTriggerStrategyBuilder {
+@Injectable()
+export class NbTriggerStrategyBuilderService {
+
   protected _host: HTMLElement;
   protected _container: () => ComponentRef<any>;
   protected _trigger: NbTrigger;
   protected _document: Document;
 
+  constructor(@Inject(NB_DOCUMENT) document) {
+    this.document(document);
+  }
+
   document(document: Document): this {
```

Comment:
I don't think we really need to keep this `document()` setter. It's used only in `constructor` above. So, let's remove it. Or better, to not introduce breaking changes, let's mark it as `@deprecated` and remove in the next version.